### PR TITLE
Improve docs for checking for the correct Form widget instance

### PR DESCRIPTION
**Pull request by mjauvin**

### CODE REVIEW

**Comment by bennothommo** [comment]
Just a thought - should we even call the `extendFields` events for nested forms? `isNested` is only `true` for nested form and repeater widgets, and it stands to reason that the user can still extend the primary form to add fields to these widgets, if necessary.

**Comment by mjauvin** [comment]
> Just a thought - should we even call the `extendFields` events for nested forms? `isNested` is only `true` for nested form and repeater widgets, and it stands to reason that the user can still extend the primary form to add fields to these widgets, if necessary.

How would that work exactly? The user would get the nested/repeater widgets from the main formWidget and extend its fields directly ?

**Comment by LukeTowers** [comment]
> Just a thought - should we even call the `extendFields` events for nested forms? `isNested` is only `true` for nested form and repeater widgets, and it stands to reason that the user can still extend the primary form to add fields to these widgets, if necessary.

That seems like a can of worms I don't want to open.

**Comment by bennothommo** [comment]
This requirement for checking `isNested` has tripped up a few people now, and I don't think it's a good experience for form changes to spill into the nested forms unless its explicit, nor is it ideal that we even have to document this whole "you need to add some condition checks on `isNested` if you don't want your nested forms to be affected."

**Comment by LukeTowers** [comment]
The problem is how people are attaching the listener. The global event fires at all times on all instances of a Form widget and it must stay that way regardless of whether or not a Form instance is nested.

It is already standard practice to check for the model class match, the controller class match, and the isNested call. I would recommend that we just provide a better example in the code example section rather than having a separate call out specifically for the isNested property.

**Comment by mjauvin** [comment]
Done @LukeTowers

File: modules/backend/widgets/Form.php [hunk 1]
```diff
@@ -500,8 +500,18 @@ protected function defineFormFields()
...
+         *             return;
+         *         }
+         *
+         *         // Only for the User model
+         *         if (!$formWidget->model instanceof \Winter\User\Models\User) {
+         *             return;
+         *         }
+         *
+         *         // Only for main form fields
+         *         if ($formWidget->isNested) {
```
Comment (LukeTowers) [inline modules/backend/widgets/Form.php]:
```suggestion
         *         // Check that we're extending the correct Form widget instance
         *         if (
         *             !($formWidget->getController() instanceof \Winter\User\Controllers\Users)
         *             || !($formWidget->model instanceof \Winter\User\Models\User)
         *             || $formWidget->isNested
         *         ) {
```

**Comment by mjauvin** [comment]
@bennothommo @LukeTowers should that be moved to 1.2.2 since it has been merged already?

**Pull request merged**